### PR TITLE
Build storage extensions against a package reference

### PR DESCRIPTION
**Pull request by pakrym**
With a project reference the packages published to a Nuget feed have a wrong dependency version.

### CODE REVIEW

**Review by kasobol-msft — APPROVED** [review]

File: eng/Directory.Build.Data.targets
Comment (pakrym) [inline eng/Directory.Build.Data.targets:123]: cc @weshaggard 

We have track 2 projects that don't start with Azure.

**Comment by pakrym** [comment]
@msftbot wait for @weshaggard  review

**Comment by pakrym** [comment]
@msftbot make sure @weshaggard has a chance to review before you merge

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [inline eng/Directory.Build.Data.targets:122]:
```suggestion
    <!-- Find all source projects -->
```

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [reply eng/Directory.Build.Data.targets:123]: OK I think this should be OK to change it should in theory work for any package reference we have a project for in the repo.

File: sdk/storage/Azure.Storage.Webjobs.Extensions.Blobs/src/Azure.WebJobs.Extensions.Storage.Blobs.csproj
Comment (weshaggard) [inline sdk/storage/Azure.Storage.Webjobs.Extensions.Blobs/src/Azure.WebJobs.Extensions.Storage.Blobs.csproj:12]: If you are depending on an unshipped version of this package it is probably better to keep it as a ProjectReference instead of pining to a given nightly build. After the version you need ships then switch to package reference.

Comment (pakrym) [reply sdk/storage/Azure.Storage.Webjobs.Extensions.Blobs/src/Azure.WebJobs.Extensions.Storage.Blobs.csproj:12]: This reference is across service directories so the pack doensn't work correctly.

File: sdk/keyvault/Microsoft.Azure.KeyVault/tests/TestFramework/Microsoft.Azure.KeyVault.TestFramework.csproj
Comment (pakrym) [inline sdk/keyvault/Microsoft.Azure.KeyVault/tests/TestFramework/Microsoft.Azure.KeyVault.TestFramework.csproj:17]: I had to add a way to opt-out for some references.

Comment (weshaggard) [reply sdk/keyvault/Microsoft.Azure.KeyVault/tests/TestFramework/Microsoft.Azure.KeyVault.TestFramework.csproj:17]: I wonder if we should go the opposite way and keep Track 2 still as "Azure" but include any other special libraries we would like to covert from package to project?

Comment (pakrym) [reply sdk/keyvault/Microsoft.Azure.KeyVault/tests/TestFramework/Microsoft.Azure.KeyVault.TestFramework.csproj:17]: It would require way more manual work to keep up-to-date than this single opt-out.

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [inline eng/Directory.Build.Data.targets:122]: It might not matter but I think this is only excluding track 1 mgmt.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**
